File: frontend/src/types/analysis.ts
export type EventAnalysis = {
  id?: string;
  event_id: string;
  summary: string;
  analysis_version?: number;
  article_count_at_analysis?: number;
  consensus_facts: Array<{ fact: string; fact_original?: string; confirmed_by: number; total: number; article_ids?: string[]; source_ids?: string[]; syndicated_count?: number }>;
  disputed_facts: Array<{ topic: string; topic_original?: string; type?: string; severity?: string; details?: unknown }>;
  blind_spots: Array<{ description: string; description_original?: string; mentioned_by?: number; total?: number }>;
  narrative_frames: Array<Record<string, unknown>>;
  timeline?: Array<{
    timestamp?: string;
    fact?: string;
    fact_original?: string;
    fragment_type?: string;
    article_id?: string;
    source_id?: string;
  }>;
  source_graph?: {
    nodes?: Array<{ id: string; label?: string; name?: string; type?: string; severity?: string }>;
    edges?: Array<{ from: string; to: string; type?: string; severity?: string; fragment_type?: string }>;
  };
};

File: backend/app/schemas/analysis.py
"""Analysis, fact fragment, and contradiction schemas."""

from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class FactFragmentRead(BaseModel):
    """Structured fact fragment returned by analysis APIs."""

    model_config = ConfigDict(from_attributes=True)

    id: UUID
    event_id: UUID
    article_id: UUID
    source_id: UUID
    fragment_type: str
    content: str
    content_en: str | None = None
    entities: dict | None = None
    numbers: dict | None = None
    source_attribution: str | None = None
    certainty_level: str | None = None
    timestamp_mentioned: datetime | None = None
    created_at: datetime


class ContradictionRead(BaseModel):
    """Contradiction returned by the API."""

    model_config = ConfigDict(from_attributes=True)

    id: UUID
    event_id: UUID
    contradiction_type: str
    description: str
    severity: str
    fragment_ids: list[UUID]
    source_ids: list[UUID]
    details: dict | None = None
    created_at: datetime


class EventAnalysisRead(BaseModel):
    """Right-panel event analysis output."""

    model_config = ConfigDict(from_attributes=True)

    id: UUID
    event_id: UUID
    summary: str
    consensus_facts: list[dict]
    disputed_facts: list[dict]
    blind_spots: list[dict]
    narrative_frames: list[dict]
    source_graph: dict | None = None
    timeline: list[dict] | None = None
    analysis_version: int
    analyzed_at: datetime
    article_count_at_analysis: int | None = None

